elrond.md

```diff
@@ -1274,6 +1330,248 @@ TODO: Implement [reserved keys and read-only runtimes](https://github.com/Elrond
+
+```
+if startingPosition < 0 || sliceLength < 0 || int(startingPosition+sliceLength) > len(sourceBytes) {
+   // does not fail execution if slice exceeds bounds
+   return 1
+}
+```
+
+
+#### Managed Buffer host functions
+
+```k
+ // extern int32_t   mBufferSetBytes(void* context, int32_t mBufferHandle, int32_t dataOffset, int32_t dataLength);
+    rule <instrs> hostCall("env", "mBufferSetBytes", [ i32 i32 i32 .ValTypes ] -> [ i32 .ValTypes ] ) 
+               => #memLoad(OFFSET, LENGTH) 
+               ~> #setBufferFromBytesStack ( ARG_IDX ) 
+               ~> #dropBytes
+               ~> i32 . const 0
+                  ... 
+         </instrs>
+         <locals> 0 |-> <i32> ARG_IDX  1 |-> <i32> OFFSET  2 |-> <i32> LENGTH </locals>
+
+ // extern int32_t   mBufferFromBigIntUnsigned(void* context, int32_t mBufferHandle, int32_t bigIntHandle);
+    rule <instrs> hostCall("env", "mBufferFromBigIntUnsigned", [ i32 i32 .ValTypes ] -> [ i32 .ValTypes ] ) 
+               => #getBigInt(BIG_IDX, Unsigned) 
+               ~> #setBufferFromBytesStack ( BUFF_IDX ) 
+               ~> #dropBytes
+               ~> i32 . const 0
+                  ... 
+         </instrs>
+         <locals> 0 |-> <i32> BUFF_IDX  1 |-> <i32> BIG_IDX </locals>
+
+ // extern int32_t   mBufferStorageStore(void* context, int32_t keyHandle, int32_t sourceHandle);
+    rule <instrs> hostCall("env", "mBufferStorageStore", [ i32 i32 .ValTypes ] -> [ i32 .ValTypes ] ) 
+               => #getBuffer(KEY_IDX) 
+               ~> #getBuffer(VAL_IDX) 
+               ~> #storageStore
+               ~> i32 . const 0
+                  ... 
+         </instrs>
+         <locals> 0 |-> <i32> KEY_IDX  1 |-> <i32> VAL_IDX </locals>
+
+ // extern int32_t   mBufferStorageLoad(void* context, int32_t keyHandle, int32_t destinationHandle);
+    rule <instrs> hostCall("env", "mBufferStorageLoad", [ i32 i32 .ValTypes ] -> [ i32 .ValTypes ] ) 
+               => #getBuffer(KEY_IDX)
+               ~> #storageLoad
+               ~> #setBufferFromBytesStack(DEST_IDX)
+               ~> #dropBytes
+               ~> i32 . const 0
+                  ... 
+         </instrs>
+         <locals> 0 |-> <i32> KEY_IDX  1 |-> <i32> DEST_IDX </locals>
+
+ // extern int32_t   mBufferToBigIntUnsigned(void* context, int32_t mBufferHandle, int32_t bigIntHandle);
+    rule <instrs> hostCall("env", "mBufferToBigIntUnsigned", [ i32 i32 .ValTypes ] -> [ i32 .ValTypes ] ) 
+               => #getBuffer(KEY_IDX)
+               ~> #setBigIntFromBytesStack(DEST_IDX, Unsigned)
+               ~> #dropBytes
+               ~> i32 . const 0
+                  ... 
+         </instrs>
+         <locals> 0 |-> <i32> KEY_IDX  1 |-> <i32> DEST_IDX </locals>
+
+ // extern int32_t   mBufferGetArgument(void* context, int32_t id, int32_t destinationHandle);
+    rule <instrs> hostCall("env", "mBufferGetArgument", [ i32 i32 .ValTypes ] -> [ i32 .ValTypes ] ) 
+               => #setBuffer(DEST_IDX, {ARGS[ARG_IDX]}:>Bytes)
```

Comment:
Yet another `lhs => #Bottom` rule, both {KItem}:>Sort and Map[KItem] are partial functions.